### PR TITLE
DEPR: Deprecate tupleize_cols in read_csv

### PR DESCRIPTION
**Pull request by gfyoung**
xref <a href="https://github.com/pandas-dev/pandas/pull/17060#issuecomment-317692667">#17060 (comment)</a>

### CODE REVIEW

File: doc/source/whatsnew/v0.21.0.txt [hunk 1]
```diff
@@ -716,6 +716,7 @@ Deprecations
 
 - :func:`read_excel()` has deprecated ``sheetname`` in favor of ``sheet_name`` for consistency with ``.to_excel()`` (:issue:`10559`).
 - :func:`read_excel()` has deprecated ``parse_cols`` in favor of ``usecols`` for consistency with :func:`read_csv` (:issue:`4988`)
+- :func:`read_csv()` has deprecated the ``tupleize_cols`` argument (:issue:`17060`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.21.0.txt]: you can say that this will always convert to a MI

**Comment by jreback** [comment]
minor comment. merge when ready.

**Comment by jorisvandenbossche** [comment]
@gfyoung The warning does not seem to appear when doing ``read_csv(..., tupleize_cols=False)``, so I think we need to put the default to None, so we can warn in those cases as well?

**Comment by gfyoung** [comment]
@jorisvandenbossche : That's deliberate.  We only issue a warning on default arguments.  Also, `tupleize_cols=False` in the context `read_csv` means return an MI, so we're perfectly fine.

File: doc/source/whatsnew/v0.21.0.txt (hunk 1 above)
Comment (gfyoung) [reply doc/source/whatsnew/v0.21.0.txt]: Yep, done.

**Comment by jorisvandenbossche** [comment]
Why do we do that deliberately? 
Nothing prevents a user to specify a default value, and eg to write `pd.read_csv(..., tupleize_cols=False)`. This code will then break in the future without any warning.

**Comment by jorisvandenbossche** [comment]
> We only issue a warning on default arguments.

You mean 'non-default' arguments?

**Comment by gfyoung** [comment]
> You mean 'non-default' arguments?

Oops, yes 😄 

> Why do we do that deliberately?
Nothing prevents a user to specify a default value, and eg to write pd.read_csv(..., tupleize_cols=False). 

That's possible.  My concern is that's going to be confusing to a user if they see `tupleize_cols=None` in the docs.  This deprecation methodology that I'm using is implemented throughout the codebase.

That being said, you do bring up an interesting point.  While I haven't seen that problem arise with any of our other deprecations, I don't see why we shouldn't discuss it.

**Comment by gfyoung** [comment]
@jorisvandenbossche : I opened #17828 to address your point about function argument deprecations.

**Comment by jorisvandenbossche** [comment]
I disagree with not raising for the default value is how it is typically done. I think it is rather the other way around (we normally put the deprecated one at None, to be able to distinguish the default and an actual user passed value), but will answer on the issue you created.

**Pull request merged**